music_assistant/providers/resonate/provider.py

```diff
@@ -66,7 +66,9 @@ async def loaded_in_mass(self) -> None:
         # Start server for handling incoming Resonate connections from clients
         # and mDNS discovery of new clients
         await self.server_api.start_server(
-            port=8927, host=cast("str", self.mass.streams.publish_ip)
+            port=8927,
+            host=self.mass.streams.bind_ip,
+            advertise_host=cast("str", self.mass.streams.publish_ip),
```

Comment:
If this cast is necessary because it could be None, better use an assert statement. That way it will break if your assumption is not correct.